### PR TITLE
Enable plugin to auto-load on IDE startup (AST-77606)

### PR DESCRIPTION
**Pull request by elchananarb**
### Description

> Enable plugin to auto-load on IDE startup

### References

> https://checkmarx.atlassian.net/browse/AST-77606

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs [hunk 1]
```diff
@@ -42,16 +44,27 @@ public static async Task InitializeAsync(AsyncPackage package)
...
+                    {
+                        throw new NotSupportedException("Cannot create tool window");
+                    }
+                    // Explicitly show the window
+                      var frame = (IVsWindowFrame)window.Frame;
+                      Microsoft.VisualStudio.ErrorHandler.ThrowOnFailure(frame.Show());
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine($"Failed to show window: {ex}");
```
Comment (miryamfoiferCX) [inline ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs:65]: should we also add this log to the new log file?

File: ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs [hunk 2]
```diff
@@ -42,16 +44,27 @@ public static async Task InitializeAsync(AsyncPackage package)
...
-                  ToolWindowPane window = await this.package.ShowToolWindowAsync(typeof(CxWindow), 0, true, this.package.DisposalToken);
-                  if ((null == window) || (null == window.Frame))
-                  {
-                      throw new NotSupportedException("Cannot create tool window");
-                  }
-              });
+            {
+                try
+                {
+                    await Task.Delay(1000);
```
Comment (miryamfoiferCX) [inline ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs:53]: why do we need this delay? better to add a comment to explain it

File: ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs (hunk 1 above)
Comment (elchananarb) [reply ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs:65]: We'll first wait for the fix related to Sara addition regarding the logs, and then there will be a task to include everything in the logs.

File: ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs (hunk 2 above)
Comment (elchananarb) [reply ast-visual-studio-extension/CxExtension/Commands/AddExtensionWindowCmd.cs:53]: done

File: ast-visual-studio-extension/CxExtension/CxWindowPackage.cs [hunk 3]
```diff
@@ -53,7 +53,10 @@ public sealed class CxWindowPackage : AsyncPackage
         protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
             try
-            { 
+            {
+                // Adding a delay of 1 second to allow the IDE to stabilize before proceeding with plugin initialization.
+                await Task.Delay(1000);
```
Comment (greensd4) [inline ast-visual-studio-extension/CxExtension/CxWindowPackage.cs:58]: didnt we have a 5 sec delay in the bootstrap? do we need another delay?

**Review by greensd4 — APPROVED** [review]

**Review by greensd4 — APPROVED** [review]

**Pull request merged**